packages/cloudflare/src/index.ts

```diff
@@ -1,34 +1,44 @@
 
-import * as fs from 'node:fs';
-import * as os from 'node:os';
-import { fileURLToPath } from 'node:url';
+import { createReadStream } from 'node:fs';
+import { appendFile, rename, stat } from 'node:fs/promises';
+import { createInterface } from 'node:readline/promises';
 import { createRedirectsFromAstroRoutes } from '@astrojs/underscore-redirects';
 import { AstroError } from 'astro/errors';
-import glob from 'tiny-glob';
 import { getPlatformProxy } from 'wrangler';
-import { deduplicatePatterns } from './utils/deduplicatePatterns.js';
+import { removeLeadingForwardSlash } from './utils/assets.js';
+import { default as createRoutesFile, getParts } from './utils/generate-routes-json.js';
 import { setImageConfig } from './utils/image-config.js';
-import { prependForwardSlash } from './utils/prependForwardSlash.js';
 import { wasmModuleLoader } from './utils/wasm-module-loader.js';
 
 export type { AdvancedRuntime } from './entrypoints/server.advanced.js';
 
 export type Options = {
 	/** Options for handling images. */
 	imageService?: 'passthrough' | 'cloudflare' | 'compile';
-	/** Configure automatic `routes.json` generation */
+	/** Configuration for `_routes.json` generation. A _routes.json file controls when your Function is invoked. This file will include three different properties:
+	 *
+	 * - version: Defines the version of the schema. Currently there is only one version of the schema (version 1), however, we may add more in the future and aim to be backwards compatible.
+	 * - include: Defines routes that will be invoked by Functions. Accepts wildcard behavior.
+	 * - exclude: Defines routes that will not be invoked by Functions. Accepts wildcard behavior. `exclude` always take priority over `include`.
+	 *
+	 * Wildcards match any number of path segments (slashes). For example, `/users/*` will match everything after the `/users/` path.
+	 *
+	 */
 	routes?: {
-		/** Strategy for generating `include` and `exclude` patterns
-		 * - `auto`: Will use the strategy that generates the least amount of entries.
-		 * - `include`: For each page or endpoint in your application that is not prerendered, an entry in the `include` array will be generated. For each page that is prerendered and whoose path is matched by an `include` entry, an entry in the `exclude` array will be generated.
-		 * - `exclude`: One `"/*"` entry in the `include` array will be generated. For each page that is prerendered, an entry in the `exclude` array will be generated.
-		 * */
-		strategy?: 'auto' | 'include' | 'exclude';
-		/** Additional `include` patterns */
-		include?: string[];
-		/** Additional `exclude` patterns */
-		exclude?: string[];
+		/** Extend `_routes.json` */
+		extend: {
+			/** Paths which should be routed to the SSR function */
+			include?: {
+				/** Generally this is in pathname format, but does support wildcards, e.g. `/users`, `/products/*` */
+				pattern: string;
```

Comment:
What's the motivation for making this an array of objects instead of an array of strings (like it was before)? It isn't explained in the PR

Reply:
@ematipico or do you think it would be better to either have an array of strings or rename the `pattern` property to `path`?

Reply:
Nope, the explanation totally makes sense! 

> or rename the pattern property to path?

If you expect to receive things like `/path/*`, then `pattern` or `glob` are better names. Maybe `pattern` fits better because I don't think you'll expect to receive actual globs e.g. `path/{foo, bar}`